### PR TITLE
Calling non_existent_method? returns NoMethodError

### PR DESCRIPTION
**Pull request by xiangge**
Calling non_existent_method? returns NoMethodError
JIRA: PDC-1962

### CODE REVIEW

**Comment by xiangge** [comment]
Some changes of the attributes_spec.rb is about fixing the pronto issue when create a new change for original files.

File: spec/pdc/resource/attributes_spec.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@
   end
 end
 
-describe PDC::Resource::Attributes do
+describe '#attributes' do
```
Comment (simozhan) [inline spec/pdc/resource/attributes_spec.rb]: Maybe we can change this to PDC::Resource::Attributes::#attributes

File: spec/pdc/resource/attributes_spec.rb [hunk 2]
```diff
@@ -27,20 +27,46 @@
...
+
+    product.wont_respond_to :foobar
+    product.must_respond_to :foobar=
+
+    product.foobar = 'bar'
+    product.foobar.must_equal 'bar'
   end
+end
 
+describe '#methods' do
```
Comment (simozhan) [inline spec/pdc/resource/attributes_spec.rb]: same with above

File: spec/pdc/resource/attributes_spec.rb [hunk 3]
```diff
@@ -27,20 +27,46 @@
...
+    product = Product.find(1)
+    assert_raises NoMethodError do
+      product.no_such_thing?
+    end
+    assert_raises NoMethodError do
+      product.no_such_thing
+    end
+  end
+end
+describe '#values' do
```
Comment (simozhan) [inline spec/pdc/resource/attributes_spec.rb]: save with above

**Comment by xiangge** [comment]
I totally agree with the rubocop changes, nice patch!
You are correct, I did many change for the pronto check, this is annoying.
Rollback codes as I includes the rubocop changes.

**Pull request merged**